backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/UserAnalysisPropertiesRepository.java

```diff
@@ -40,4 +41,9 @@ public void setUserProperties(String configurationScopeId, Map<String, String> e
     propertiesByConfigScope.put(configurationScopeId, new HashMap<>(extraProperties));
   }
 
+  public void setOrUpdatePathToCompileCommands(String configurationScopeId, String value) {
+    var properties = propertiesByConfigScope.getOrDefault(configurationScopeId, new HashMap<>());
+    properties.put(PATH_TO_COMPILE_COMMANDS, value);
+    setUserProperties(configurationScopeId, properties);
```

Comment:
What if the user properties are set again after by the client? Shouldn't we store this in a separate field?

Reply:
Changed implementation.